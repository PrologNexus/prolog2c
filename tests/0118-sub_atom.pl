main :-
	sub_atom(hello, B, L, A, X),
	writeq([B, X, L, A]), nl,
	fail.


sub_atom(X, _, _, _, _) :-
	var(X), throw(instantiation_error).
sub_atom(ATOM, BEFORE, LEN, AFTER, SATOM) :-
	atom_length(ATOM, ALEN),
	ALEN2 is ALEN - 1,
	between(0, ALEN, BEFORE),
	between(0, ALEN, AFTER),
	LEN is ALEN - AFTER - BEFORE,
	LEN >= 0,
	foreign_call(sub_atom(ATOM, BEFORE, LEN, SATOM)).
