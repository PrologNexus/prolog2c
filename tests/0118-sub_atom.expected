[0,hello,5,0]
[0,hell,4,1]
[0,hel,3,2]
[0,he,2,3]
[0,h,1,4]
[0,'',0,5]
[1,ello,4,0]
[1,ell,3,1]
[1,el,2,2]
[1,e,1,3]
[1,'',0,4]
[2,llo,3,0]
[2,ll,2,1]
[2,l,1,2]
[2,'',0,3]
[3,lo,2,0]
[3,l,1,1]
[3,'',0,2]
[4,o,1,0]
[4,'',0,1]
[5,'',0,0]
